yarn-project/bb-prover/src/bb/cli.ts

```diff
@@ -70,6 +71,7 @@ export function getProgram(log: LogFn): Command {
     .requiredOption('-b, --bb-path <string>', 'The path to the BB binary', BB_BINARY_PATH)
     .requiredOption('-c, --circuit <string>', 'The name of a protocol circuit')
     .requiredOption('-n --contract-name <string>', 'The name of the contract to generate', 'contract.sol')
+    .option('-r, --recursive', 'Whether a SNARK friendly key should be generated', false)
```

Comment:
verifier contracts are never recursion friendly so this flag doesn't make sense.

This is because non-recursive proofs are generated using the keccak256 hash function. We need to use the same hash function during verification so proofs generated using keccak256 are cheap to verify on ethereum (where the verifier contract would be deployed) whereas if we want to verify the proof within a snark (for recursion) we want to use a snark-friendly hash function.

Reply:
Right, that makes sense, thanks 👍